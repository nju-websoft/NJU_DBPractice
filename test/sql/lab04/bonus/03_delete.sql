open database db2025;

-- Test DELETE performance and correctness with various index scenarios
-- Indexes available: stock_idx(s_i_id), stock_swid_idx(s_w_id, s_i_id), etc.

-- 1. Test single column equality delete (should use stock_idx)
delete from stock where s_i_id = 150;
select s_i_id, s_quantity from stock where s_i_id = 150;

-- 2. Test range delete using primary index
delete from stock where s_i_id > 250 and s_i_id < 255;
select s_i_id, s_w_id from stock where s_i_id > 249 and s_i_id < 256 order by s_i_id;

-- 3. Test delete by composite index (should use stock_swid_idx)
delete from stock where s_w_id = 5 and s_i_id = 350;
select s_i_id, s_w_id from stock where s_w_id = 5 and s_i_id = 350;

-- 4. Test composite index with range (should use stock_swid_idx)
delete from stock where s_w_id = 10 and s_i_id > 400 and s_i_id < 405;
select s_i_id, s_w_id from stock where s_w_id = 10 and s_i_id > 399 and s_i_id < 406 order by s_i_id;

-- 5. Test delete by quantity index (should use stock_sqty_idx)
delete from stock where s_quantity = 85 and s_i_id > 500 and s_i_id < 510;
select s_i_id, s_quantity from stock where s_quantity = 85 and s_i_id > 499 and s_i_id < 511 order by s_i_id;

-- 6. Test range delete on quantity (should use stock_sqty_idx)
delete from stock where s_quantity > 95 and s_quantity < 98;
select s_i_id, s_quantity from stock where s_quantity > 94 and s_quantity < 99 order by s_i_id;

-- 7. Test delete using order count index
delete from stock where s_order_cnt = 75 and s_i_id > 600 and s_i_id < 610;
select s_order_cnt, s_i_id from stock where s_order_cnt = 75 and s_i_id > 599 and s_i_id < 611 order by s_i_id;

-- 8. Test three-column composite index delete
delete from stock where s_order_cnt = 50 and s_remote_cnt = 25 and s_ytd > 5.0 and s_ytd < 6.0;
select s_order_cnt, s_remote_cnt, s_ytd from stock where s_order_cnt = 50 and s_remote_cnt = 25 and s_ytd > 4.9 and s_ytd < 6.1 order by s_i_id;

-- 9. Test delete by ytd range
delete from stock where s_ytd > 9.0 and s_ytd < 9.5;
select s_ytd, s_i_id from stock where s_ytd > 8.9 and s_ytd < 9.6 order by s_i_id;

-- 10. Test complex multi-condition delete
delete from stock where s_w_id = 15 and s_quantity > 80 and s_order_cnt > 60;
select s_w_id, s_quantity, s_order_cnt from stock where s_w_id = 15 and s_quantity > 79 and s_order_cnt > 59 order by s_i_id;

-- 11. Test boundary value deletes
delete from stock where s_i_id = 0;
delete from stock where s_i_id = 9999;
select s_i_id from stock where s_i_id = 0;
select s_i_id from stock where s_i_id = 9999;

-- 12. Test non-indexed column delete (should use table scan)
delete from stock where s_dist_01 = 'ABC123';
select s_dist_01 from stock where s_dist_01 = 'ABC123' order by s_i_id;

-- 13. Test delete affecting multiple records
delete from stock where s_w_id = 20;
select s_w_id, s_i_id from stock where s_w_id = 20 order by s_i_id;

-- 14. Test overlapping index conditions
delete from stock where s_i_id > 700 and s_i_id < 710 and s_quantity = 50;
select s_i_id, s_quantity from stock where s_i_id > 699 and s_i_id < 711 and s_quantity = 50 order by s_i_id;

-- 15. Test delete with float precision
delete from stock where s_ytd = 7.5;
select s_ytd from stock where s_ytd = 7.5 order by s_i_id;
