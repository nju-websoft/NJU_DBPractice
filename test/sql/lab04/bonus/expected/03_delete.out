
+--------------+
| deleted      | 
+--------------+
| 1            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_quantity   | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 4            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_w_id       | 
+--------------+--------------+
| 250          | 875          | 
+--------------+--------------+
| 255          | 417          | 
+--------------+--------------+
Total tuple(s): 2

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_w_id       | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_w_id       | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_quantity   | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 179          | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_quantity   | 
+--------------+--------------+
| 30           | 98           | 
+--------------+--------------+
| 66           | 98           | 
+--------------+--------------+
| 70           | 98           | 
+--------------+--------------+
| 131          | 95           | 
+--------------+--------------+
| 351          | 95           | 
+--------------+--------------+
| 490          | 98           | 
+--------------+--------------+
| 493          | 98           | 
+--------------+--------------+
| 540          | 98           | 
+--------------+--------------+
| 552          | 95           | 
+--------------+--------------+
| 664          | 98           | 
+--------------+--------------+
| 699          | 95           | 
+--------------+--------------+
| 718          | 98           | 
+--------------+--------------+
| 730          | 95           | 
+--------------+--------------+
| 807          | 98           | 
+--------------+--------------+
| 812          | 98           | 
+--------------+--------------+
| 873          | 98           | 
+--------------+--------------+
| 935          | 95           | 
+--------------+--------------+
| 990          | 95           | 
+--------------+--------------+
| 1166         | 95           | 
+--------------+--------------+
| 1178         | 98           | 
+--------------+--------------+
| 1311         | 98           | 
+--------------+--------------+
| 1352         | 98           | 
+--------------+--------------+
| 1506         | 95           | 
+--------------+--------------+
| 1571         | 98           | 
+--------------+--------------+
| 1629         | 98           | 
+--------------+--------------+
| 1647         | 98           | 
+--------------+--------------+
| 1801         | 95           | 
+--------------+--------------+
| 1981         | 98           | 
+--------------+--------------+
| 1995         | 95           | 
+--------------+--------------+
| 2192         | 95           | 
+--------------+--------------+
| 2207         | 95           | 
+--------------+--------------+
| 2238         | 98           | 
+--------------+--------------+
| 2259         | 95           | 
+--------------+--------------+
| 2310         | 98           | 
+--------------+--------------+
| 2319         | 98           | 
+--------------+--------------+
| 2338         | 95           | 
+--------------+--------------+
| 2378         | 98           | 
+--------------+--------------+
| 2441         | 95           | 
+--------------+--------------+
| 2455         | 98           | 
+--------------+--------------+
| 2470         | 95           | 
+--------------+--------------+
| 2504         | 95           | 
+--------------+--------------+
| 2527         | 98           | 
+--------------+--------------+
| 2596         | 98           | 
+--------------+--------------+
| 2727         | 98           | 
+--------------+--------------+
| 2774         | 98           | 
+--------------+--------------+
| 2794         | 98           | 
+--------------+--------------+
| 2803         | 95           | 
+--------------+--------------+
| 2816         | 98           | 
+--------------+--------------+
| 2822         | 98           | 
+--------------+--------------+
| 2832         | 95           | 
+--------------+--------------+
| 2939         | 95           | 
+--------------+--------------+
| 3003         | 95           | 
+--------------+--------------+
| 3004         | 98           | 
+--------------+--------------+
| 3055         | 98           | 
+--------------+--------------+
| 3071         | 98           | 
+--------------+--------------+
| 3218         | 95           | 
+--------------+--------------+
| 3280         | 98           | 
+--------------+--------------+
| 3309         | 95           | 
+--------------+--------------+
| 3413         | 98           | 
+--------------+--------------+
| 3417         | 95           | 
+--------------+--------------+
| 3419         | 95           | 
+--------------+--------------+
| 3431         | 98           | 
+--------------+--------------+
| 3651         | 95           | 
+--------------+--------------+
| 3665         | 95           | 
+--------------+--------------+
| 3707         | 95           | 
+--------------+--------------+
| 3738         | 98           | 
+--------------+--------------+
| 3816         | 95           | 
+--------------+--------------+
| 3842         | 98           | 
+--------------+--------------+
| 3886         | 95           | 
+--------------+--------------+
| 3945         | 98           | 
+--------------+--------------+
| 4033         | 98           | 
+--------------+--------------+
| 4086         | 95           | 
+--------------+--------------+
| 4161         | 98           | 
+--------------+--------------+
| 4171         | 98           | 
+--------------+--------------+
| 4199         | 95           | 
+--------------+--------------+
| 4366         | 98           | 
+--------------+--------------+
| 4390         | 95           | 
+--------------+--------------+
| 4411         | 98           | 
+--------------+--------------+
| 4419         | 98           | 
+--------------+--------------+
| 4485         | 95           | 
+--------------+--------------+
| 4527         | 95           | 
+--------------+--------------+
| 4543         | 98           | 
+--------------+--------------+
| 4617         | 95           | 
+--------------+--------------+
| 4652         | 95           | 
+--------------+--------------+
| 4690         | 98           | 
+--------------+--------------+
| 4701         | 98           | 
+--------------+--------------+
| 4820         | 98           | 
+--------------+--------------+
| 4923         | 95           | 
+--------------+--------------+
| 4966         | 98           | 
+--------------+--------------+
| 4998         | 95           | 
+--------------+--------------+
| 5008         | 95           | 
+--------------+--------------+
| 5014         | 98           | 
+--------------+--------------+
| 5022         | 98           | 
+--------------+--------------+
| 5055         | 98           | 
+--------------+--------------+
| 5100         | 95           | 
+--------------+--------------+
| 5153         | 98           | 
+--------------+--------------+
| 5175         | 95           | 
+--------------+--------------+
| 5272         | 95           | 
+--------------+--------------+
| 5312         | 95           | 
+--------------+--------------+
| 5317         | 95           | 
+--------------+--------------+
| 5328         | 98           | 
+--------------+--------------+
| 5394         | 95           | 
+--------------+--------------+
| 5480         | 98           | 
+--------------+--------------+
| 5551         | 95           | 
+--------------+--------------+
| 5573         | 98           | 
+--------------+--------------+
| 5585         | 95           | 
+--------------+--------------+
| 5661         | 95           | 
+--------------+--------------+
| 5666         | 98           | 
+--------------+--------------+
| 5776         | 95           | 
+--------------+--------------+
| 5790         | 95           | 
+--------------+--------------+
| 5806         | 98           | 
+--------------+--------------+
| 5830         | 95           | 
+--------------+--------------+
| 5836         | 98           | 
+--------------+--------------+
| 5841         | 95           | 
+--------------+--------------+
| 5882         | 98           | 
+--------------+--------------+
| 5966         | 98           | 
+--------------+--------------+
| 5986         | 95           | 
+--------------+--------------+
| 6109         | 95           | 
+--------------+--------------+
| 6121         | 98           | 
+--------------+--------------+
| 6150         | 95           | 
+--------------+--------------+
| 6187         | 95           | 
+--------------+--------------+
| 6200         | 95           | 
+--------------+--------------+
| 6344         | 98           | 
+--------------+--------------+
| 6379         | 95           | 
+--------------+--------------+
| 6409         | 98           | 
+--------------+--------------+
| 6411         | 98           | 
+--------------+--------------+
| 6458         | 98           | 
+--------------+--------------+
| 6489         | 95           | 
+--------------+--------------+
| 6515         | 95           | 
+--------------+--------------+
| 6524         | 95           | 
+--------------+--------------+
| 6548         | 98           | 
+--------------+--------------+
| 6594         | 95           | 
+--------------+--------------+
| 6619         | 95           | 
+--------------+--------------+
| 6699         | 98           | 
+--------------+--------------+
| 6789         | 95           | 
+--------------+--------------+
| 6810         | 98           | 
+--------------+--------------+
| 6859         | 95           | 
+--------------+--------------+
| 6935         | 95           | 
+--------------+--------------+
| 6980         | 95           | 
+--------------+--------------+
| 6983         | 95           | 
+--------------+--------------+
| 7014         | 98           | 
+--------------+--------------+
| 7046         | 98           | 
+--------------+--------------+
| 7048         | 98           | 
+--------------+--------------+
| 7054         | 98           | 
+--------------+--------------+
| 7057         | 98           | 
+--------------+--------------+
| 7113         | 98           | 
+--------------+--------------+
| 7174         | 98           | 
+--------------+--------------+
| 7187         | 98           | 
+--------------+--------------+
| 7283         | 98           | 
+--------------+--------------+
| 7317         | 98           | 
+--------------+--------------+
| 7332         | 98           | 
+--------------+--------------+
| 7426         | 95           | 
+--------------+--------------+
| 7452         | 95           | 
+--------------+--------------+
| 7532         | 98           | 
+--------------+--------------+
| 7603         | 95           | 
+--------------+--------------+
| 7624         | 95           | 
+--------------+--------------+
| 7676         | 98           | 
+--------------+--------------+
| 7732         | 95           | 
+--------------+--------------+
| 7758         | 95           | 
+--------------+--------------+
| 7768         | 95           | 
+--------------+--------------+
| 7794         | 98           | 
+--------------+--------------+
| 7801         | 98           | 
+--------------+--------------+
| 7875         | 95           | 
+--------------+--------------+
| 7897         | 95           | 
+--------------+--------------+
| 7952         | 98           | 
+--------------+--------------+
| 8020         | 98           | 
+--------------+--------------+
| 8030         | 98           | 
+--------------+--------------+
| 8092         | 98           | 
+--------------+--------------+
| 8118         | 95           | 
+--------------+--------------+
| 8200         | 95           | 
+--------------+--------------+
| 8235         | 95           | 
+--------------+--------------+
| 8326         | 98           | 
+--------------+--------------+
| 8441         | 95           | 
+--------------+--------------+
| 8455         | 98           | 
+--------------+--------------+
| 8456         | 95           | 
+--------------+--------------+
| 8515         | 98           | 
+--------------+--------------+
| 8530         | 98           | 
+--------------+--------------+
| 8558         | 95           | 
+--------------+--------------+
| 8828         | 95           | 
+--------------+--------------+
| 8939         | 98           | 
+--------------+--------------+
| 8953         | 95           | 
+--------------+--------------+
| 8958         | 98           | 
+--------------+--------------+
| 9012         | 95           | 
+--------------+--------------+
| 9035         | 98           | 
+--------------+--------------+
| 9044         | 98           | 
+--------------+--------------+
| 9134         | 98           | 
+--------------+--------------+
| 9144         | 95           | 
+--------------+--------------+
| 9165         | 98           | 
+--------------+--------------+
| 9255         | 95           | 
+--------------+--------------+
| 9427         | 95           | 
+--------------+--------------+
| 9440         | 95           | 
+--------------+--------------+
| 9513         | 98           | 
+--------------+--------------+
| 9537         | 98           | 
+--------------+--------------+
| 9578         | 95           | 
+--------------+--------------+
| 9636         | 98           | 
+--------------+--------------+
| 9682         | 95           | 
+--------------+--------------+
| 9793         | 95           | 
+--------------+--------------+
| 9887         | 98           | 
+--------------+--------------+
| 9911         | 95           | 
+--------------+--------------+
| 9916         | 98           | 
+--------------+--------------+
Total tuple(s): 200

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_order_cnt  | s_i_id       | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+--------------+
| s_order_cnt  | s_remote_cnt | s_ytd        | 
|              |              |              | 
+--------------+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 462          | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_ytd        | s_i_id       | 
+--------------+--------------+
| 9.578416     | 26           | 
+--------------+--------------+
| 8.994043     | 51           | 
+--------------+--------------+
| 9.503351     | 87           | 
+--------------+--------------+
| 8.958344     | 96           | 
+--------------+--------------+
| 8.915624     | 110          | 
+--------------+--------------+
| 9.552320     | 111          | 
+--------------+--------------+
| 9.526227     | 120          | 
+--------------+--------------+
| 8.969028     | 198          | 
+--------------+--------------+
| 8.997320     | 360          | 
+--------------+--------------+
| 9.588724     | 409          | 
+--------------+--------------+
| 9.583426     | 413          | 
+--------------+--------------+
| 8.995164     | 481          | 
+--------------+--------------+
| 9.563952     | 533          | 
+--------------+--------------+
| 9.554617     | 605          | 
+--------------+--------------+
| 9.532629     | 655          | 
+--------------+--------------+
| 8.979856     | 703          | 
+--------------+--------------+
| 9.541723     | 709          | 
+--------------+--------------+
| 9.541055     | 732          | 
+--------------+--------------+
| 9.546080     | 738          | 
+--------------+--------------+
| 9.502205     | 797          | 
+--------------+--------------+
| 8.978326     | 823          | 
+--------------+--------------+
| 9.524264     | 1007         | 
+--------------+--------------+
| 8.903759     | 1048         | 
+--------------+--------------+
| 9.583484     | 1064         | 
+--------------+--------------+
| 9.533511     | 1118         | 
+--------------+--------------+
| 8.954692     | 1140         | 
+--------------+--------------+
| 9.578362     | 1143         | 
+--------------+--------------+
| 9.512360     | 1215         | 
+--------------+--------------+
| 8.931460     | 1249         | 
+--------------+--------------+
| 8.958468     | 1346         | 
+--------------+--------------+
| 9.529450     | 1382         | 
+--------------+--------------+
| 8.986787     | 1437         | 
+--------------+--------------+
| 8.911025     | 1442         | 
+--------------+--------------+
| 8.997408     | 1480         | 
+--------------+--------------+
| 9.588687     | 1527         | 
+--------------+--------------+
| 9.537550     | 1538         | 
+--------------+--------------+
| 9.559168     | 1666         | 
+--------------+--------------+
| 8.909777     | 1729         | 
+--------------+--------------+
| 8.975088     | 1888         | 
+--------------+--------------+
| 9.539150     | 1935         | 
+--------------+--------------+
| 9.583218     | 2014         | 
+--------------+--------------+
| 8.950284     | 2119         | 
+--------------+--------------+
| 8.954280     | 2163         | 
+--------------+--------------+
| 8.925815     | 2185         | 
+--------------+--------------+
| 9.599655     | 2197         | 
+--------------+--------------+
| 8.985254     | 2212         | 
+--------------+--------------+
| 9.526526     | 2233         | 
+--------------+--------------+
| 8.968922     | 2285         | 
+--------------+--------------+
| 8.987602     | 2293         | 
+--------------+--------------+
| 8.977587     | 2512         | 
+--------------+--------------+
| 8.906275     | 2522         | 
+--------------+--------------+
| 9.542421     | 2690         | 
+--------------+--------------+
| 8.924520     | 2733         | 
+--------------+--------------+
| 9.517017     | 2743         | 
+--------------+--------------+
| 8.964543     | 2790         | 
+--------------+--------------+
| 8.938416     | 2898         | 
+--------------+--------------+
| 8.971298     | 2938         | 
+--------------+--------------+
| 8.911062     | 3007         | 
+--------------+--------------+
| 8.951146     | 3074         | 
+--------------+--------------+
| 9.547987     | 3214         | 
+--------------+--------------+
| 8.920983     | 3279         | 
+--------------+--------------+
| 8.918988     | 3309         | 
+--------------+--------------+
| 8.962042     | 3363         | 
+--------------+--------------+
| 9.501645     | 3376         | 
+--------------+--------------+
| 9.576668     | 3383         | 
+--------------+--------------+
| 9.560664     | 3408         | 
+--------------+--------------+
| 8.903389     | 3462         | 
+--------------+--------------+
| 8.984425     | 3540         | 
+--------------+--------------+
| 9.556264     | 3583         | 
+--------------+--------------+
| 9.595710     | 3591         | 
+--------------+--------------+
| 9.568115     | 3611         | 
+--------------+--------------+
| 9.519425     | 3622         | 
+--------------+--------------+
| 8.980988     | 3685         | 
+--------------+--------------+
| 9.532111     | 3782         | 
+--------------+--------------+
| 9.520971     | 3823         | 
+--------------+--------------+
| 8.980914     | 3849         | 
+--------------+--------------+
| 8.948868     | 3869         | 
+--------------+--------------+
| 8.979103     | 3935         | 
+--------------+--------------+
| 8.921323     | 4062         | 
+--------------+--------------+
| 9.556832     | 4080         | 
+--------------+--------------+
| 8.904633     | 4099         | 
+--------------+--------------+
| 8.909472     | 4148         | 
+--------------+--------------+
| 8.995417     | 4209         | 
+--------------+--------------+
| 9.519030     | 4251         | 
+--------------+--------------+
| 9.569512     | 4254         | 
+--------------+--------------+
| 8.984618     | 4261         | 
+--------------+--------------+
| 9.504436     | 4292         | 
+--------------+--------------+
| 9.543073     | 4293         | 
+--------------+--------------+
| 8.935574     | 4326         | 
+--------------+--------------+
| 9.556191     | 4419         | 
+--------------+--------------+
| 8.959826     | 4470         | 
+--------------+--------------+
| 8.953336     | 4566         | 
+--------------+--------------+
| 8.926598     | 4575         | 
+--------------+--------------+
| 8.926547     | 4612         | 
+--------------+--------------+
| 9.575452     | 4625         | 
+--------------+--------------+
| 8.993028     | 4632         | 
+--------------+--------------+
| 8.968149     | 4633         | 
+--------------+--------------+
| 9.533041     | 4636         | 
+--------------+--------------+
| 9.537473     | 4639         | 
+--------------+--------------+
| 8.950632     | 4730         | 
+--------------+--------------+
| 8.989356     | 4791         | 
+--------------+--------------+
| 8.978415     | 4877         | 
+--------------+--------------+
| 8.971786     | 4947         | 
+--------------+--------------+
| 8.948250     | 4974         | 
+--------------+--------------+
| 8.942591     | 4987         | 
+--------------+--------------+
| 8.935431     | 5018         | 
+--------------+--------------+
| 8.927527     | 5075         | 
+--------------+--------------+
| 8.995317     | 5096         | 
+--------------+--------------+
| 9.574497     | 5156         | 
+--------------+--------------+
| 8.983848     | 5392         | 
+--------------+--------------+
| 8.994799     | 5461         | 
+--------------+--------------+
| 9.526095     | 5481         | 
+--------------+--------------+
| 9.577735     | 5595         | 
+--------------+--------------+
| 8.950745     | 5608         | 
+--------------+--------------+
| 8.983980     | 5638         | 
+--------------+--------------+
| 8.900203     | 5645         | 
+--------------+--------------+
| 9.572144     | 5677         | 
+--------------+--------------+
| 9.548354     | 5847         | 
+--------------+--------------+
| 8.982668     | 5852         | 
+--------------+--------------+
| 8.947538     | 5897         | 
+--------------+--------------+
| 8.917190     | 5898         | 
+--------------+--------------+
| 8.994760     | 5962         | 
+--------------+--------------+
| 9.568921     | 5975         | 
+--------------+--------------+
| 9.589410     | 6006         | 
+--------------+--------------+
| 8.917886     | 6063         | 
+--------------+--------------+
| 9.585895     | 6122         | 
+--------------+--------------+
| 8.968628     | 6179         | 
+--------------+--------------+
| 8.960440     | 6209         | 
+--------------+--------------+
| 8.972021     | 6239         | 
+--------------+--------------+
| 9.561220     | 6241         | 
+--------------+--------------+
| 9.522156     | 6389         | 
+--------------+--------------+
| 9.544039     | 6537         | 
+--------------+--------------+
| 9.575914     | 6538         | 
+--------------+--------------+
| 9.538817     | 6555         | 
+--------------+--------------+
| 8.930729     | 6586         | 
+--------------+--------------+
| 8.932096     | 6588         | 
+--------------+--------------+
| 9.532339     | 6667         | 
+--------------+--------------+
| 8.971844     | 6675         | 
+--------------+--------------+
| 8.988389     | 6721         | 
+--------------+--------------+
| 9.523042     | 6787         | 
+--------------+--------------+
| 9.558818     | 6838         | 
+--------------+--------------+
| 8.949360     | 6886         | 
+--------------+--------------+
| 8.931869     | 6914         | 
+--------------+--------------+
| 9.540545     | 6927         | 
+--------------+--------------+
| 8.999180     | 6975         | 
+--------------+--------------+
| 9.598166     | 6976         | 
+--------------+--------------+
| 8.910703     | 6979         | 
+--------------+--------------+
| 9.518991     | 6980         | 
+--------------+--------------+
| 8.983713     | 6984         | 
+--------------+--------------+
| 8.901222     | 6996         | 
+--------------+--------------+
| 9.559238     | 7000         | 
+--------------+--------------+
| 8.964053     | 7072         | 
+--------------+--------------+
| 9.529092     | 7217         | 
+--------------+--------------+
| 9.594902     | 7307         | 
+--------------+--------------+
| 9.539716     | 7342         | 
+--------------+--------------+
| 8.974060     | 7344         | 
+--------------+--------------+
| 9.526349     | 7436         | 
+--------------+--------------+
| 9.512142     | 7451         | 
+--------------+--------------+
| 8.925726     | 7464         | 
+--------------+--------------+
| 9.517746     | 7621         | 
+--------------+--------------+
| 9.596721     | 7663         | 
+--------------+--------------+
| 9.519547     | 7681         | 
+--------------+--------------+
| 9.552589     | 7700         | 
+--------------+--------------+
| 8.912786     | 7770         | 
+--------------+--------------+
| 8.969307     | 7868         | 
+--------------+--------------+
| 8.974665     | 7883         | 
+--------------+--------------+
| 9.535796     | 7995         | 
+--------------+--------------+
| 9.566842     | 8022         | 
+--------------+--------------+
| 9.565321     | 8061         | 
+--------------+--------------+
| 8.961236     | 8100         | 
+--------------+--------------+
| 8.938647     | 8226         | 
+--------------+--------------+
| 8.919698     | 8252         | 
+--------------+--------------+
| 8.960810     | 8279         | 
+--------------+--------------+
| 8.976319     | 8306         | 
+--------------+--------------+
| 9.502590     | 8308         | 
+--------------+--------------+
| 8.988571     | 8334         | 
+--------------+--------------+
| 9.552066     | 8492         | 
+--------------+--------------+
| 8.993749     | 8506         | 
+--------------+--------------+
| 8.939813     | 8509         | 
+--------------+--------------+
| 8.951486     | 8563         | 
+--------------+--------------+
| 8.996757     | 8574         | 
+--------------+--------------+
| 9.555385     | 8693         | 
+--------------+--------------+
| 8.919200     | 8709         | 
+--------------+--------------+
| 9.592447     | 8716         | 
+--------------+--------------+
| 8.925912     | 8752         | 
+--------------+--------------+
| 9.521522     | 8771         | 
+--------------+--------------+
| 9.567801     | 8900         | 
+--------------+--------------+
| 9.529140     | 8902         | 
+--------------+--------------+
| 8.936790     | 8976         | 
+--------------+--------------+
| 8.991349     | 9010         | 
+--------------+--------------+
| 8.943975     | 9027         | 
+--------------+--------------+
| 9.515731     | 9031         | 
+--------------+--------------+
| 8.950195     | 9032         | 
+--------------+--------------+
| 9.538651     | 9033         | 
+--------------+--------------+
| 9.568604     | 9209         | 
+--------------+--------------+
| 9.505802     | 9227         | 
+--------------+--------------+
| 9.552138     | 9323         | 
+--------------+--------------+
| 9.518244     | 9348         | 
+--------------+--------------+
| 8.902564     | 9367         | 
+--------------+--------------+
| 9.553570     | 9426         | 
+--------------+--------------+
| 8.926664     | 9453         | 
+--------------+--------------+
| 8.978691     | 9493         | 
+--------------+--------------+
| 9.570820     | 9521         | 
+--------------+--------------+
| 9.580558     | 9544         | 
+--------------+--------------+
| 9.543473     | 9614         | 
+--------------+--------------+
| 8.978505     | 9683         | 
+--------------+--------------+
| 8.933497     | 9810         | 
+--------------+--------------+
| 9.514708     | 9826         | 
+--------------+--------------+
| 9.554649     | 9900         | 
+--------------+--------------+
| 9.592146     | 9930         | 
+--------------+--------------+
| 8.964727     | 9950         | 
+--------------+--------------+
| 9.587481     | 9980         | 
+--------------+--------------+
| 8.924179     | 9989         | 
+--------------+--------------+
| 9.548510     | 9996         | 
+--------------+--------------+
| 8.938025     | 9997         | 
+--------------+--------------+
Total tuple(s): 215

+--------------+
| deleted      | 
+--------------+
| 1            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+--------------+
| s_w_id       | s_quantity   | s_order_cnt  | 
+--------------+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 1            | 
+--------------+
Total tuple(s): 1

+--------------+
| deleted      | 
+--------------+
| 1            | 
+--------------+
Total tuple(s): 1

+--------------+
| s_i_id       | 
+--------------+
Total tuple(s): 0

+--------------+
| s_i_id       | 
+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+
| s_dist_01    | 
+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 10           | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_w_id       | s_i_id       | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+--------------+
| s_i_id       | s_quantity   | 
+--------------+--------------+
Total tuple(s): 0

+--------------+
| deleted      | 
+--------------+
| 0            | 
+--------------+
Total tuple(s): 1

+--------------+
| s_ytd        | 
+--------------+
Total tuple(s): 0
